File: src/main/java/com/mcyzj/lib/plugin/JiangPlugin.java
package com.mcyzj.lib.plugin;

import com.mcyzj.lib.bukkit.menu.MenuListener;
import org.bukkit.Bukkit;
import org.bukkit.plugin.java.JavaPlugin;

public class JiangPlugin extends JavaPlugin {
    public static JiangPlugin jiangPlugin = null;
    @Override
    public void onEnable() {
        JiangLib.loadLibs();
        Bukkit.getPluginManager().registerEvents(new MenuListener(), this);
        enable();
    }

    public void enable() {}

    @Override
    public void onDisable() {
        disable();
    }

    public void disable() {}
}

File: src/main/kotlin/com/mcyzj/lib/plugin/JiangPlugin.kt
package com.mcyzj.lib.plugin

import org.bukkit.plugin.java.JavaPlugin

abstract class JiangPlugin: JavaPlugin() {
    open fun load() {}

    open fun enable() {}

    open fun disable() {}

    override fun onLoad() {
        load()
    }

    override fun onEnable() {
        JiangLib.loadLibs()
        enable()
    }

    override fun onDisable() {
        disable()
    }
}